File: 2016/Prototypes/ShooterTestProgram/src/org/usfirst/frc/team3081/robot/commands/ShooterCommand.java
package org.usfirst.frc.team3081.robot.commands;

import org.usfirst.frc.team3081.robot.Robot;

import edu.wpi.first.wpilibj.command.Command;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

/**
 *
 */
public class ShooterCommand extends Command {

    public ShooterCommand() {
        // Use requires() here to declare subsystem dependencies
        // eg. requires(chassis);
    	requires(Robot.shooterSubsystem);
    	SmartDashboard.putNumber("Left", 0.0);
    	SmartDashboard.putNumber("Right", 0.0);

    }

    // Called just before this Command runs the first time
    protected void initialize() {
    	Robot.shooterSubsystem.stop();
    }

    // Called repeatedly when this Command is scheduled to run
    protected void execute() {
    	double left = SmartDashboard.getNumber("Left", 0.0);
    	double right = SmartDashboard.getNumber("Right", 0.0);
    	System.out.printf("Left %f, Right %f\n", left, right);
    	Robot.shooterSubsystem.set(left, right);
    }

    // Make this return true when this Command no longer needs to run execute()
    protected boolean isFinished() {
        return false;
    }

    // Called once after isFinished returns true
    protected void end() {
    	Robot.shooterSubsystem.stop();
    }

    // Called when another command which requires one or more of the same
    // subsystems is scheduled to run
    protected void interrupted() {
    	Robot.shooterSubsystem.stop();
    }
}

File: 2017/2017Robot/src/Commands/ShooterCommand.cpp
#include "ShooterCommand.h"

ShooterCommand::ShooterCommand() {
	// Use Requires() here to declare subsystem dependencies
	// eg. Requires(Robot::chassis.get());
	Requires(shooterSubsystem.get());
}

// Called just before this Command runs the first time
void ShooterCommand::Initialize() {
	shooterSubsystem->Stop();
}

// Called repeatedly when this Command is scheduled to run
void ShooterCommand::Execute() {
	shooterSubsystem->Shoot();
}

// Make this return true when this Command no longer needs to run execute()
bool ShooterCommand::IsFinished() {
	return false;
}

// Called once after isFinished returns true
void ShooterCommand::End() {
	shooterSubsystem->Stop();
}

// Called when another command which requires one or more of the same
// subsystems is scheduled to run
void ShooterCommand::Interrupted() {
	shooterSubsystem->Stop();
}
